### ISSUE TITLE
Can we have ability to switch providers and models?

### ISSUE THREAD

**Issue by kneelesh48** [BuilderIO/ai-shell#130]
I'd like to have a few providers like OpenAI, Anthropic, Groq and  models like gpt-4o, Claude Sonnet, and Qwen-2.5-Coder.

Is it possible to save all there providers in a config file and switch when needed?

**Comment by thepandatimes** [comment]
Yes please. I would also like to have something like ollama support or maybe mistral even.

**Comment by trevorstr** [comment]
I'd like to be able to use the xAI Grok and Amazon Bedrock APIs with this.

I don't have a subscription to OpenAI.

**Comment by lasantosr** [comment]
Might want to have a look at [intelli-shell](https://github.com/lasantosr/intelli-shell) (disclaimer: i'm the author)
